### PR TITLE
Hacktoberfest update

### PR DESCRIPTION
**Pull request by burtonr**
## Description

Updates to the Hacktoberfest feature from the comments on the original PR #134 

## Motivation and Context

- [X] I have raised an issue to propose this change ([required](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md))
Some suggestions were made on the original PR as well as in Slack.

## How Has This Been Tested?

Deployed Derek from my repo: `burtonr/derek:pr-hacktoberfest7`
Currently running on https://github.com/burtonr/testing

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I've read the [CONTRIBUTION](https://github.com/alexellis/derek/blob/master/CONTRIBUTING.md) guide
- [X] I have signed-off my commits with `git commit -s`
- [X] I have added tests to cover my changes.
- [X] All new and existing tests passed.

### CODE REVIEW

**Comment by burtonr** [comment]
This is a draft as I still need to update the documentation.

More importantly, I cannot figure out how or why the "sign-off" comment/check is still being run...

Screenshot from 2019-10-10 00-14-14

It seems no matter what happens, the sign-off comment is added. Perhaps it's a race where the "isSpam" is not yet returned, but the next feature is run

**Comment by jonatasbaldin** [comment]
Should we make a time check to see if the PR date is in October? We'd probably lower the false positives 🤔

File: handler/hacktoberfest_handler.go [hunk 1]
```diff
@@ -56,11 +56,19 @@ func HandleHacktoberfestPR(req types.PullRequestOuter, contributingURL string, c
...
 
-			return
+			return true, nil
 		}
 	}
+	return false, nil
+}
+
+func hacktoberfestSpamComment(contributingURL string) string {
+	return `Thank you for your contribution. I've checked and your commit does not appear to follow the guidelines in our [contributing guide](` + contributingURL + `). Spelling and README changes are better handled by opening an issue.
```
Comment (alexellis) [inline handler/hacktoberfest_handler.go]: Instead:

> Thank you for your interest in this project.

**Comment by burtonr** [comment]
Found the cause of the multiple comments/labels was the individual events on the pull request by each of the actions performed by the hacktoberfest handler. Add label, close, add comment.

To prevent the additional handlers from firing, I've re-arranged the hacktoberfest handler to **first** close the PR, then do the other things. In addition, I've added the `State` to the Pull Request type so that the `Action` and `State` are verified to not be "closed" before calling the handlers.

Link to the repo used for testing this: https://github.com/burtonr/more-bots-2

> Note: The image used for these tests was created before updating the message to say "Thanks for your interest...". The message has since been updated.

"Spam" PR:

Screenshot from 2019-10-13 23-19-49

Readme change with signed off commit:

Screenshot from 2019-10-13 23-27-36

Code file change without signoff:

Screenshot from 2019-10-13 23-30-09

**Pull request merged**